src/shared/components/Modal.tsx

```diff
@@ -0,0 +1,223 @@
+                    sx={{ color, width: "99%" }}
+                    className="edit-modal-content"
+                    placeholder="Add a description..."
+                    defaultValue={content}
+                    onChange={handleChange}
+                  />
+                </Box>
+                {modalMedia || preview ? (
+                  <Box
+                    sx={{
+                      mt: "20px",
+                      display: "flex",
+                      justifyContent: "center",
+                      flexDirection: "column",
+                    }}
+                  >
+                    <Box
+                      component="img"
+                      src={modalMedia ? getSecureUrl(modalMedia) : preview}
+                      sx={{
+                        width: "100%",
+                        paddingTop: "5px",
+                        m: "auto",
+                      }}
+                    />
+                    <Button
+                      sx={{
+                        m: "auto",
+                        width: "200px",
+                        color: "red",
+                        textTransform: "none",
+                        "&:hover": { backgroundColor: navButtons },
+                        mt: "10px",
+                        fontSize: "18px",
+                      }}
+                      onClick={() => {
+                        modalMedia ? setModalMedia(null) : onRemove();
+                      }}
+                    >
+                      Remove Image
+                    </Button>
+                  </Box>
+                ) : (
+                  <Box>
+                    <Box
+                      display="flex"
+                      sx={{ width: "30%", justifyContent: "space-between" }}
+                    >
+                      <label htmlFor="file-upload-modal">
+                        <Box
+                          sx={{
+                            color: color,
+                            display: "flex",
+                            justifyContent: "center",
+                            alignItems: "center",
+                            pr: "15px",
+                            ":hover": {
+                              cursor: "pointer",
+                              backgroundColor: navButtons,
+                            },
+                            mt: "20px",
+                          }}
+                        >
+                          <InsertPhotoOutlined sx={{ color }} />
+                          <Typography sx={{ fontSize: "12px", px: "2px" }}>
+                            Media
+                          </Typography>
+                        </Box>
+                      </label>
+                      <Input
+                        id="file-upload-modal"
+                        type="file"
+                        sx={{ display: "none" }}
+                        onChange={onSelectFile}
+                      />
```

Comment:
The id of the Input had to be different from the sharePost and shareComment for the modal to display the media being uploaded.